src/operations.cc

```diff
@@ -392,4 +392,46 @@ namespace sharp {
     return image.colourspace(VIPS_INTERPRETATION_B_W) >= threshold;
   }
 
+  VImage Trim(VImage image, int const threshold) {
+    // An equivalent of ImageMagick's -trim in C++ ... automatically remove
+    // "boring" image edges.
+
+    // We use .project to sum the rows and columns of a 0/255 mask image, the first
+    // non-zero row or column is the object edge. We make the mask image with an
+    // amount-different-from-background image plus a threshold.
+
+    // find the value of the pixel at (0, 0) ... we will search for all pixels
+    // significantly different from this
+    std::vector<double> background = image(0, 0);
```

Comment:
Maybe we should let the users define the the point from where the trimming color is picked. Don't know how we should handle this (by a color or a position). For now it's always the top-left color.